### ISSUE TITLE
Repair armor/all/hand permissions do not work correctly

### ISSUE THREAD

**Issue by Frostified** [Zrips/CMI#3575]
**Description of issue:**
Repair armor/all/hand permissions do not work
---

**Cmi Version (using`/cmi version`):** 
8.7.0.1 latest
**Server Type (Spigot/Paperspigot/etc):** 
Spigot
**Server Version (using `/ver`):** 
1.16.1

**Comment by Zrips** [comment]
As as for keepexp issue, check if player has access to permission with /cmi haspermission command. Just to make sure that its recognized properly before we start trying to find out whats causing it

**Comment by Frostified** [comment]
> As as for keepexp issue, check if player has access to permission with /cmi haspermission command. Just to make sure that its recognized properly before we start trying to find out whats causing it

Will do.

**Comment by Frostified** [comment]
Player has access to the permissions. Upon running the command i receive "cmi.command.repair.hand set to true by some plugin"
Yet the players with permissions for repairing hands,armor, or all, cannot access the given permissions.

**Comment by Frostified** [comment]
> As as for keepexp issue, check if player has access to permission with /cmi haspermission command. Just to make sure that its recognized properly before we start trying to find out whats causing it

Confirmed as broken by testing with alts and your permissions checkers.

**Comment by Zrips** [comment]
Did you tested this one with CMI and permission plugin only? As i did one more testing, everything works as it should, player keeps its exp if he has access to permission node. This was done on a clean server with minimal amount of plugins

**Comment by Frostified** [comment]
> Did you tested this one with CMI and permission plugin only? As i did one more testing, everything works as it should, player keeps its exp if he has access to permission node. This was done on a clean server with minimal amount of plugins

This is not a bug for experience. And is still confirmed as broken.

**Comment by Zrips** [comment]
Did you tested without any plugin but CMI and permission plugin?

**Comment by Frostified** [comment]
> Did you tested without any plugin but CMI and permission plugin?

Yes sir. Just your CMI plugin and PermissionsEX. Still didnt work :(

**Comment by Frostified** [comment]
Can we get a fix on this?

**Comment by Zrips** [comment]
I tested this couple times already. Tho it did tested it with LuckPerms. By default player needs to have cmi.command.repair permission node, after that he will need cmi.command.repair.hand if he wants to repair hand items, which can be seen with error message while hovering over it. After that will depend which variable he will use, all or armor for which again, permission is requested. I cant really fix something what is not broken or atleast what i cant reproduce. Try testing this feature with different permission plugin like LuckPerms, as your PEX could have some invalid setup or something.

**Comment by Frostified** [comment]
> I tested this couple times already. Tho it did tested it with LuckPerms. By default player needs to have cmi.command.repair permission node, after that he will need cmi.command.repair.hand if he wants to repair hand items, which can be seen with error message while hovering over it. After that will depend which variable he will use, all or armor for which again, permission is requested. I cant really fix something what is not broken or atleast what i cant reproduce. Try testing this feature with different permission plugin like LuckPerms, as your PEX could have some invalid setup or something.

Proof its not working:
https://youtu.be/MoONNp2KtP0

I dont want to switch permissions plugin unless absoultely necessary.

**Comment by Frostified** [comment]
New update:
Secondary commands such as "/repair all" or "/repair hand" do not work for players that are not OP

cmi.command.repair - If the player has this permission they can use "/repair" but it only repairs the item in their hand. Not items in their inventory.

cmi.command.repair.hand - If the player has this permission they can use "/repair" but again it only repairs the item in their hand. Not items in their inventory. Does not work with "/repair hand"

cmi.command.repair.all - if the player has this permission they can use "/repair" but it repairs everything in their inventory. Does not work with "/repair all"

cmi.command.repair.armor - if the player has this permission it says you do not have access to this.

All these permissions were tested individually, not combined together in any way . 

How am i supposed to setup cooldowns for hand/armor/all

**Comment by Frostified** [comment]
"/repair" targets multiple events depending on the permission the player has

**Comment by Frostified** [comment]
Cooldowns do not work either.

**Comment by the456gamer** [comment]
Unable to reproduce with:
```
[23:37:50 INFO]: --------------------------------------------------
[23:37:50 INFO]: CMI: 8.7.2.6 SqLite
[23:37:50 INFO]: Server: Paper(93) 1.16.1-R0.1-SNAPSHOT
[23:37:50 INFO]: CMI economy: False CMI Chat: False 
[23:37:50 INFO]: Modules -> 50 enabled 1 disabled: votifier
[23:37:50 INFO]: --------------------------------------------------
```
and
```
[23:38:18 INFO]: Plugins (4): CMI, CMIPaperLib, LuckPerms, TesterPlugin
```
<details><summary>Screenshots</summary>
<p>

Setting permissions
![image](https://user-images.githubusercontent.com/42748092/87836350-ab66c280-c887-11ea-89a0-9be81ab2a2c6.png)

Giving a diamond sword with missing durability:
command used: `give the456gamer diamond_sword{Damage:1000} 9`
![image](https://i.imgur.com/rZXhJSQ.png)

![image](https://i.imgur.com/rZXhJSQ.png)
![image](https://i.imgur.com/zrkFRSS.png)
![image](https://i.imgur.com/1k0400E.png)
![image](https://i.imgur.com/EmcdGpV.png)
![image](https://i.imgur.com/7ofYZCY.png)
![image](https://i.imgur.com/WLHPVW3.png)
![image](https://i.imgur.com/0Y3cApE.png)
![image](https://i.imgur.com/a8lUE65.png)
![image](https://i.imgur.com/SJr1U6A.png)
![image](https://i.imgur.com/dxuDIU8.png)
![image](https://i.imgur.com/WnrFWfR.png)

</p>
</details>
as far as i can see, all working as intented

**Comment by Frostified** [comment]
> Unable to reproduce with:
> 
> ```
> [23:37:50 INFO]: --------------------------------------------------
> [23:37:50 INFO]: CMI: 8.7.2.6 SqLite
> [23:37:50 INFO]: Server: Paper(93) 1.16.1-R0.1-SNAPSHOT
> [23:37:50 INFO]: CMI economy: False CMI Chat: False 
> [23:37:50 INFO]: Modules -> 50 enabled 1 disabled: votifier
> [23:37:50 INFO]: --------------------------------------------------
> ```
> 
> and
> 
> ```
> [23:38:18 INFO]: Plugins (4): CMI, CMIPaperLib, LuckPerms, TesterPlugin
> ```
> 
> Screenshots
> as far as i can see, all working as intended

Im not using paper. Or CMIpaper LIB or tester plugin, or luck perms. Your results are irrelevant along with your attitude on discord. Get a life

**Comment by Frostified** [comment]
To make this more clear:
- You cannot target /repair all or /repair hand correctly. /repair armor is the only one working as intended

**Comment by the456gamer** [comment]
did you look at the screenshots? i clearly show /cmi repair hand, and /cmi repair all in use, and it correctly targeting the correct items

**Comment by the456gamer** [comment]
can you do /ver permissionex ? i want to try it with the same setup as you

**Comment by Frostified** [comment]
> did you look at the screenshots? i clearly show /cmi repair hand, and /cmi repair all in use, and it correctly targeting the correct items
> edit: also im not oped
Anyone can post screenshots repairing items in their hand. Show a video of what i just said working.

**Comment by the456gamer** [comment]
i currently dont have easy acces to obs, but ill try

**Comment by Frostified** [comment]
> i currently dont have easy acces to obs, but ill try

And your not op and dont have '*"?
This is being tested on "player" accounts

**Comment by the456gamer** [comment]
yes (well its a test server, but not special permissions)

**Comment by Frostified** [comment]
I have video of it NOT WORKING. So i dont beleive you unless i see video

**Comment by the456gamer** [comment]
so what pex version are you using?
unstable beta (v2)
or old out of date unsupported (v1)
https://dev.bukkit.org/projects/permissionsex (old)
https://jenkins.addstar.com.au/job/PermissionsEx/ (dev build)

**Comment by Frostified** [comment]
Issue was fixed without anyone telling me even though i have it on video not working. Real nice way to fix a bug.

**Comment by mrfloris** [comment]
<img width="423" alt="Screenshot 2020-07-18 at 02 51 42" src="https://user-images.githubusercontent.com/28841349/87840795-c8a88a80-c8a1-11ea-8c50-944b1041f772.png">

This implies in the video that another plugin is preventing chat, I don't know what plugin is doing that, and if that includes commands. But might explain why commands are also not working. 

The video also shows text from other plugins, so the claim it's just CMI is something I cannot get behind (from watching the video). 

But I am glad to hear you have it resolved. 

In regards to the comment about attitude on Discord, you were getting courtesy support from people who didn't have to, you were actually the one with the name calling, swearing, and the attitude. Please refrain from doing so ~ thanks for understanding.

**Comment by Frostified** [comment]
> <img alt="Screenshot 2020-07-18 at 02 51 42" width="423" src="https://user-images.githubusercontent.com/28841349/87840795-c8a88a80-c8a1-11ea-8c50-944b1041f772.png">
> 
> This implies in the video that another plugin is preventing chat, I don't know what plugin is doing that, and if that includes commands. But might explain why commands are also not working.
> 
> The video also shows text from other plugins, so the claim it's just CMI is something I cannot get behind (from watching the video).
> 
> But I am glad to hear you have it resolved.
> 
> In regards to the comment about attitude on Discord, you were getting courtesy support from people who didn't have to, you were actually the one with the name calling, swearing, and the attitude. Please refrain from doing so ~ thanks for understanding.

Another member in the community helped me out because your so called golden boy is an arguing piece of crap. He didnt move either when he logged in and helped me make sure its working in the latest plugin. So to say its not CMI is you just being in denial and not accepting facts. I can sit here and go all day. I have video proof its not working, i have not changed my setup and now its suddenly working? You dont care if the issue was resolved you just dont like helping people. I posted 3 times about an announcement thing and a member of the community helped me, NOT YOU. In regards to you getting involved on discord your friend doesnt know how to help someone without using the same setup or claiming thats it working for him. This was closed but i guess its not now. Wanna keep arguing over how you guys fixed a bug without confirming it as fixed? I literally have video proof of it being broken without changing my setup one bit. And your actually a piece of shit MrFloris so go fuck yourself thanks for understanding. Your involvement was uncalled for and you have no idea what was going on nor do you look at chat history.

**Comment by Frostified** [comment]
I will never post a bug report again. Ill just leave this plugin riddled with its 900+ reports. Terrible experience
